File: mobile-app/lib/core/utils/phone_utils.dart
class PhoneUtils {
  static String clean(String raw) {
    var p = raw.replaceAll(RegExp(r'[\s\-\(\)]'), '');
    if (p.startsWith('+255')) p = p.substring(4);
    else if (p.startsWith('255')) p = p.substring(3);
    if (p.startsWith('0')) p = p.substring(1);
    return p;
  }
  static String display(String phone) =>
      '+255 ${phone.substring(0,3)} ${phone.substring(3,6)} ${phone.substring(6)}';
}

File: lib/utils/phoneUtils.ts
export function cleanPhone(raw: string): string {
  let p = raw.replace(/[\s\-\(\)]/g, '')
  if (p.startsWith('+255')) p = p.slice(4)
  else if (p.startsWith('255')) p = p.slice(3)
  if (p.startsWith('0')) p = p.slice(1)
  return p
}

export function formatPhoneDisplay(phone: string): string {
  return `+255 ${phone.slice(0,3)} ${phone.slice(3,6)} ${phone.slice(6)}`
}
